internal/decomposer/decomposer.go

```diff
@@ -157,6 +163,15 @@ func (decomposer *Decomposer) ExecuteStep(step cacao.Step, scopeVariables cacao.
 		return decomposer.actionExecutor.Execute(metadata, actionMetadata)
 	case cacao.StepTypePlaybookAction:
 		return decomposer.playbookActionExecutor.Execute(metadata, step, variables)
+	case cacao.StepTypeIfCondition:
+		stepId, branch, err := decomposer.conditionExecutor.Execute(metadata, step, variables)
+		if err != nil {
+			return cacao.NewVariables(), err
+		}
+		if branch {
+			return decomposer.ExecuteBranch(stepId, variables)
+		}
```

Comment:
If on_completion is handled at decomposer level, then I think its invocation should be performed here

Reply:
We are not branching if the on_true or on_false are not taken but for consistency sake we return the onCompletionId